velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -267,76 +266,76 @@ public boolean callCMD(Object player, String cmd) {
 
   @Override
   public String parseJSON(InputStreamReader json, String key) {
-    JsonElement element = new JsonParser().parse(json);
-    if (element instanceof JsonNull) {
+    JsonObject element = (JsonObject) JsonParser.parseReader(json);
+    if (element.isJsonNull()) {
```

Comment:
One thing or another is wrong here. Either:
* `element` is always a JsonObject, in which case it will never be a JsonNull.
* `element` can be a JsonNull, in which case the cast to JsonObject is premature, and will cause ClassCastException.

The isJsonXXX methods perform simple instanceof checks - arguably a poor decision from the Gson team. I'd recommend using `instanceof JsonNull` rather than `isJsonNull()`, for readability reasons.

Reply:
the main reason behind this is because IntelliJ complains and says the two types cannot be casted. Instead, I've made a try-catch for getAsJsonObject() which in turn throws IllegalStateException where I can then return null.